backend/src/test/java/com/now/naaga/place/service/PlaceServiceTest.java

```diff
@@ -0,0 +1,73 @@
+package com.now.naaga.place.service;
+
+import com.now.naaga.common.builder.PlaceBuilder;
+import com.now.naaga.common.fixture.PositionFixture;
+import com.now.naaga.place.application.PlaceService;
+import com.now.naaga.place.application.dto.RecommendPlaceCommand;
+import com.now.naaga.place.domain.Place;
+import com.now.naaga.place.domain.Position;
+import com.now.naaga.place.exception.PlaceException;
+import org.junit.jupiter.api.RepeatedTest;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.transaction.annotation.Transactional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.SoftAssertions.assertSoftly;
+
+@ActiveProfiles("test")
+@Sql("/truncate.sql")
+@SpringBootTest
+public class PlaceServiceTest {
+
+    @Autowired
+    PlaceService placeService;
+
+    @Autowired
+    PlaceBuilder placeBuilder;
+
+    @Transactional
+    @RepeatedTest(value = 10)
+    void 사용자로부터_300미터_이내의_장소_추천한다() {
+        //given
+        final Position 현재위치 = PositionFixture.잠실_루터회관_정문_근처_좌표;
+
+        final Position 잠실역교보문고좌표 = PositionFixture.잠실역_교보문고_좌표;
+        final Position 잠실루터회관정문좌표 = PositionFixture.잠실_루터회관_정문_좌표;
+        final Position 제주좌표 = PositionFixture.제주_좌표;
+
+        final Place 잠실역교보문고 = placeBuilder.init()
+                .position(잠실역교보문고좌표)
+                .build();
+        final Place 잠실루터회관정문 = placeBuilder.init()
+                .position(잠실루터회관정문좌표)
+                .build();
+        final Place 제주 = placeBuilder.init()
+                .position(제주좌표)
+                .build();
+
+        final RecommendPlaceCommand recommendPlaceCommand = new RecommendPlaceCommand(현재위치);
+
+        //when
+        final Place place = placeService.recommendPlaceByPosition(recommendPlaceCommand);
+
+        //then
+        assertSoftly(softAssertions -> {
+            assertThat(place).usingRecursiveComparison().isNotEqualTo(제주);
+            assertThat(place).usingRecursiveComparison().isIn(잠실루터회관정문, 잠실역교보문고);
+        });
+    }
+    @Test
+    void 사용자로부터_300미터_이내의_추천할_장소가_없으면_예외를_발생한다() {
+        //given
+        final Position 현재위치 = PositionFixture.잠실_루터회관_정문_근처_좌표;
+        final RecommendPlaceCommand recommendPlaceCommand = new RecommendPlaceCommand(현재위치);
+
+        // when & then
+        assertThatThrownBy(() -> placeService.recommendPlaceByPosition(recommendPlaceCommand)).isInstanceOf(PlaceException.class);
```

Comment:
P3:

```suggestion
        // when
        final BaseExceptionType baseExceptionType = assertThrows(PlaceException.class, 
        () -> placeService.createPlace(createPlaceCommand)).exceptionType();

        // then
        assertThat(baseExceptionType).isEqualTo(PlaceExceptionType.ALREADY_EXIST_NEARBY);
```

예외에 대한 검증은 JUnit5 의 `assertThrows()` 메서드를 활용하면 더 자세한 검증을 할 수 있습니다 👨‍🚀

Reply:
asserThorws로 검증하면 그 타입을 받을 수 있군요!!
적극 수용하겠습니다.
다만 when절로 따로 분리하지 않고 현제처럼 웬, 덴 절로 assertSoftly로 묶었습니다. 저 assertThrows자체가 검증 작업도 하기 때문에 예외를 발생시키지 않으면 그 뒤에 검증에도 영향이 끼치니까 하나로 묶을게요